recipe/build.sh

```diff
@@ -4,17 +4,62 @@ set -euxo pipefail
 
 if [[ "${target_platform}" == osx-* ]]; then
   export LDFLAGS="${LDFLAGS} -lz -framework CoreFoundation -Xlinker -undefined -Xlinker dynamic_lookup"
+  # Remove stdlib=libc++; this is the default and errors on C sources.
+  export CXXFLAGS=${CXXFLAGS//-stdlib=libc++}
 else
   export LDFLAGS="${LDFLAGS} -lrt"
 fi
+export CFLAGS="${CFLAGS} -DNDEBUG"
+export CXXFLAGS="${CXXFLAGS} -DNDEBUG"
 
-export LD_LIBRARY_PATH="${LD_LIBRARY_PATH+LD_LIBRARY_PATH:}:$PREFIX/lib"
+export BUILD_FLAGS="--target_cpu_features default --enable_mkl_dnn"
+
+#  - if JAX_RELEASE or JAXLIB_RELEASE are set: version looks like "0.4.16"
+#  - if JAX_NIGHTLY or JAXLIB_NIGHTLY are set: version looks like "0.4.16.dev20230906"
+#  - if none are set: version looks like "0.4.16.dev20230906+ge58560fdc
+export JAXLIB_RELEASE=1
+
+if [[ ${cuda_compiler_version} != "None" ]]; then
+  export HERMETIC_CUDA_COMPUTE_CAPABILITIES=sm_60,sm_70,sm_75,sm_80,sm_86,sm_89,sm_90,compute_90
+  export CUDA_HOME="${BUILD_PREFIX}/targets/x86_64-linux"
+  export PATH=$PATH:${BUILD_PREFIX}/nvvm/bin
+ 
+  # XLA can only cope with a single cuda header include directory, merge both
+  rsync -a ${PREFIX}/targets/x86_64-linux/include/ ${BUILD_PREFIX}/targets/x86_64-linux/include/
+
+  # Although XLA supports a non-hermetic build, it still tries to find headers in the hermetic locations.
+  # We do this in the BUILD_PREFIX to not have any impact on the resulting jaxlib package.
+  # Otherwise, these copied files would be included in the package.
+  rm -rf ${BUILD_PREFIX}/targets/x86_64-linux/include/third_party
+  mkdir -p ${BUILD_PREFIX}/targets/x86_64-linux/include/third_party/gpus/cuda/extras/CUPTI
+  cp -r ${PREFIX}/targets/x86_64-linux/include ${BUILD_PREFIX}/targets/x86_64-linux/include/third_party/gpus/cuda/
+  cp -r ${PREFIX}/targets/x86_64-linux/include ${BUILD_PREFIX}/targets/x86_64-linux/include/third_party/gpus/cuda/extras/CUPTI/
+  mkdir -p ${BUILD_PREFIX}/targets/x86_64-linux/include/third_party/gpus/cudnn
+  cp ${PREFIX}/include/cudnn.h ${BUILD_PREFIX}/targets/x86_64-linux/include/third_party/gpus/cudnn/
+
+  export LOCAL_CUDA_PATH="${BUILD_PREFIX}/targets/x86_64-linux"
+  export LOCAL_CUDNN_PATH="${PREFIX}/targets/x86_64-linux"
+  export LOCAL_NCCL_PATH="${PREFIX}/targets/x86_64-linux"
+  export TF_CUDA_VERSION="${cuda_compiler_version}"
+  export TF_CUDNN_VERSION="8"
```

Comment:
is there a way to remove this magic number? it could be missed when cudnn is upgraded to 9. Also, why is cudnn v8 being used? 9.1.1.17 is the latest